state/mongodb/mongodb.go

```diff
@@ -195,10 +206,28 @@ func (m *MongoDB) setInternal(ctx context.Context, req *state.SetRequest) error
 		filter[etag] = uuid.NewString()
 	}
 
-	update := bson.M{"$set": bson.M{id: req.Key, value: v, etag: uuid.NewString()}}
-	_, err := m.collection.UpdateOne(ctx, filter, update, options.Update().SetUpsert(true))
+	reqTTL, err := stateutils.ParseTTL(req.Metadata)
+	if err != nil {
+		return fmt.Errorf("failed to parse TTL: %w", err)
+	}
 
-	return err
+	var ttl any
+	if reqTTL != nil {
+		ttl = primitive.NewDateTimeFromTime(time.Now().Add(time.Second * time.Duration(*reqTTL)))
```

Comment:
Most other components when they set the TTL, use the time from the DB server. The problem of using the time from the app like here is that if there are multiple apps accessing the data, and there's a clock skew (not that uncommon), this may cause undefined behavior. Can the TTL be set in a relative way, such as "10 seconds from now" so MongoDB uses its own clock?

Reply:
Done